delta-api/src/main/java/io/cdap/delta/api/DMLEvent.java

```diff
@@ -121,6 +122,22 @@ public boolean equals(Object o) {
       Objects.equals(sortKeys, dmlEvent.sortKeys);
   }
 
+  @Override
+  public String toString() {
+    StringJoiner stringJoiner = new StringJoiner(", ", "{", "}");
+    stringJoiner.add("operation=" + operation);
+    stringJoiner.add("isSnapshot=" + this.isSnapshot());
+    stringJoiner.add("changeType=" + this.getChangeType());
+    stringJoiner.add("offset=" + this.getOffset());
```

Comment:
Is offset guaranteed to be not null ?

Reply:
offset can be null in that case it returns an empty map.